diff --git a/.gitignore b/.gitignore
index 34272350b..91c64e337 100644
--- a/.gitignore
+++ b/.gitignore
@@ -47,6 +47,7 @@ test/string-test
 test/tags-test
 test/thread-test
 test/url-test
+fuzz/address-fuzz
 
 # Build products
 *.o
diff --git a/Makefile.autosetup b/Makefile.autosetup
index 3e97becf9..868078ac6 100644
--- a/Makefile.autosetup
+++ b/Makefile.autosetup
@@ -443,7 +443,7 @@ uninstall: $(UNINSTALL_TARGETS)
 distclean: clean
 	$(RM) $(DEPFILES) .clang_complete autosetup/jimsh0 config.h config.log \
 		conststrings.c contrib/Makefile doc/Makefile doc/neomutt.1 \
-		html Makefile po/Makefile test/Makefile
+		html Makefile po/Makefile test/Makefile fuzz/Makefile
 	$(RM) *.gc?? */*.gc?? test/*/*.gc??
 	$(RM) lcov.info lcov
 
@@ -474,5 +474,8 @@ coverage: all test
 @if ENABLE_UNIT_TESTS
 @include @srcdir@/test/Makefile.autosetup
 @endif
+@if ENABLE_FUZZ_TESTS
+@include @srcdir@/fuzz/Makefile.autosetup
+@endif
 
 # vim: set ts=8 noexpandtab:
diff --git a/auto.def b/auto.def
index 95e55c05b..16e7a9009 100644
--- a/auto.def
+++ b/auto.def
@@ -18,7 +18,7 @@ define PACKAGE_VERSION  "20180716"
 define BUGS_ADDRESS     "neomutt-devel@neomutt.org"
 
 # Subdirectories that contain additional Makefile.autosetup files
-set subdirs {po doc contrib}
+set subdirs {po doc contrib fuzz}
 ###############################################################################
 
 ###############################################################################
@@ -95,6 +95,8 @@ options {
 # Testing
   testing=0                 => "Enable Unit Testing"
   coverage=0                => "Enable Coverage Testing"
+# Fuzzing
+  fuzzing                   => "Enable Fuzz Testing"
 # Configure with pkg-config
   pkgconf=0                 => "Use pkg-config during configure"
 # Enable all options
@@ -109,7 +111,7 @@ options {
 if {1} {
   # Keep sorted, please.
   foreach opt {
-    bdb backtrace coverage doc everything fmemopen full-doc gdbm gnutls gpgme
+    bdb backtrace coverage doc everything fmemopen full-doc fuzzing gdbm gnutls gpgme
     gss homespool idn idn2 inotify kyotocabinet lmdb locales-fix lua mixmaster
     nls notmuch pkgconf pgp qdbm sasl smime ssl testing tokyocabinet
   } {
@@ -541,6 +543,13 @@ if {[get-define want-testing]} {
   lappend subdirs test
 }
 
+###############################################################################
+# Fuzz Testing
+if {[get-define want-fuzzing]} {
+  define ENABLE_FUZZ_TESTS
+  lappend subdirs fuzz
+}
+
 ###############################################################################
 # Lua
 if {[get-define want-lua]} {
diff --git a/main.c b/main.c
index b451faaa7..5e13f4c2a 100644
--- a/main.c
+++ b/main.c
@@ -403,7 +403,13 @@ bool get_user_info(struct ConfigSet *cs)
  * @retval 0 Success
  * @retval 1 Error
  */
-int main(int argc, char *argv[], char *envp[])
+int
+#ifdef ENABLE_FUZZ_TESTS
+disabled_main
+#else
+main
+#endif
+(int argc, char *argv[], char *envp[])
 {
   char *subject = NULL;
   char *include_file = NULL;
