diff --git c/Makefile.autosetup w/Makefile.autosetup
index ebd9fe59b..539dfbaaf 100644
--- c/Makefile.autosetup
+++ w/Makefile.autosetup
@@ -525,6 +525,7 @@ distclean: clean
 		test/Makefile .clang_complete
 	$(RM) *.gc?? */*.gc?? test/*/*.gc??
 	$(RM) lcov.info lcov
+	$(RM) zzz.inc doxygen-build.txt tmp.txt
 
 @if ENABLE_COVERAGE
 # coverage testing
@@ -540,6 +541,11 @@ coverage: all test
 	lcov --list lcov.info
 @endif
 
+# Web Code docs (doxygen)
+.PHONY: html $(PWD)/mutt
+html:
+	$(SRCDIR)/doxygen/generate.sh
+
 ##############################################################################
 # include generated dependency files
 -include $(DEPFILES)
diff --git c/email/lib.h w/email/lib.h
index c5b4fc776..73b8182b2 100644
--- c/email/lib.h
+++ w/email/lib.h
@@ -41,6 +41,8 @@
  * | email/tags.c           | @subpage email_tags      |
  * | email/thread.c         | @subpage email_thread    |
  * | email/url.c            | @subpage email_url       |
+ *
+ * @image html libemail.svg
  */
 
 #ifndef MUTT_EMAIL_LIB_H
diff --git c/mutt/lib.h w/mutt/lib.h
index 8dfa7dd91..9e52f3536 100644
--- c/mutt/lib.h
+++ w/mutt/lib.h
@@ -54,6 +54,8 @@
  *
  * @note The library is self-contained -- some files may depend on others in
  *       the library, but none depends on source from outside.
+ *
+ * @image html libmutt.svg
  */
 
 #ifndef MUTT_MUTT_LIB_H
