From db82113637529952bf62923690bd4b672ed50c0e Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Sun, 29 Jul 2018 17:27:21 +0100
Subject: [PATCH] travis: add doxygen config

---
 Makefile.autosetup | 6 ++++++
 email/email.h      | 2 ++
 mutt/mutt.h        | 2 ++
 3 files changed, 10 insertions(+)

diff --git a/Makefile.autosetup b/Makefile.autosetup
index f6005f0ae..4e9fc1971 100644
--- a/Makefile.autosetup
+++ b/Makefile.autosetup
@@ -422,10 +422,16 @@ distclean: clean
 	$(RM) $(DEPFILES) .clang_complete autosetup/jimsh0 config.h config.log \
 		conststrings.c contrib/Makefile doc/Makefile doc/neomutt.1 \
 		html Makefile po/Makefile test/Makefile
+	$(RM) zzz.inc doxygen-build.txt tmp.txt
 
 # Tests for the config code
 config-test: libmutt.a libemail.a libconfig.a
 
+# Web Code docs (doxygen)
+.PHONY: html $(PWD)/mutt
+html:
+	$(SRCDIR)/doxygen/generate.sh
+
 ##############################################################################
 # include generated dependency files
 -include $(DEPFILES)
diff --git a/email/email.h b/email/email.h
index 0c70db626..0c71e6a70 100644
--- a/email/email.h
+++ b/email/email.h
@@ -43,6 +43,8 @@
  * | email/tags.c           | @subpage email_tags      |
  * | email/thread.c         | @subpage email_thread    |
  * | email/url.c            | @subpage email_url       |
+ *
+ * @image html libemail.svg
  */
 
 #ifndef _EMAIL_EMAIL_H
diff --git a/mutt/mutt.h b/mutt/mutt.h
index b8b771302..bd4425f00 100644
--- a/mutt/mutt.h
+++ b/mutt/mutt.h
@@ -49,6 +49,8 @@
  *
  * @note The library is self-contained -- some files may depend on others in
  *       the library, but none depends on source from outside.
+ *
+ * @image html libmutt.svg
  */
 
 #ifndef _MUTT_MUTT_H
