From 90c685925b0b854d0f28d9c2e6766e923bb8e594 Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Thu, 27 Sep 2018 11:07:48 +0100
Subject: [PATCH] doxy

---
 Makefile.autosetup | 6 ++++++
 email/lib.h        | 2 ++
 mutt/mutt.h        | 2 ++
 3 files changed, 10 insertions(+)

diff --git a/Makefile.autosetup b/Makefile.autosetup
index 697134f1c..3a05af8c2 100644
--- a/Makefile.autosetup
+++ b/Makefile.autosetup
@@ -425,10 +425,16 @@ distclean: clean
 	$(RM) $(DEPFILES) .clang_complete autosetup/jimsh0 config.h config.log \
 		conststrings.c contrib/Makefile doc/Makefile doc/neomutt.1 \
 		html Makefile po/Makefile test/Makefile
+	$(RM) zzz.inc doxygen-build.txt tmp.txt
 
 # Tests for the config code
 config-test: libmutt.a libemail.a libconfig.a
 
+# Web Code docs (doxygen)
+.PHONY: html $(PWD)/mutt
+html:
+	$(SRCDIR)/doxygen/generate.sh
+
 ##############################################################################
 # include generated dependency files
 -include $(DEPFILES)
diff --git a/email/lib.h b/email/lib.h
index ef1a20a8a..5794d4111 100644
--- a/email/lib.h
+++ b/email/lib.h
@@ -43,6 +43,8 @@
  * | email/tags.c           | @subpage email_tags      |
  * | email/thread.c         | @subpage email_thread    |
  * | email/url.c            | @subpage email_url       |
+ *
+ * @image html libemail.svg
  */
 
 #ifndef MUTT_EMAIL_LIB_H
diff --git a/mutt/mutt.h b/mutt/mutt.h
index ffc53041f..9a0eb6b29 100644
--- a/mutt/mutt.h
+++ b/mutt/mutt.h
@@ -50,6 +50,8 @@
  *
  * @note The library is self-contained -- some files may depend on others in
  *       the library, but none depends on source from outside.
+ *
+ * @image html libmutt.svg
  */
 
 #ifndef MUTT_LIB_MUTT_H
