diff --git a/Makefile.autosetup b/Makefile.autosetup
index 2ba78677a..0fedab3f5 100644
--- a/Makefile.autosetup
+++ b/Makefile.autosetup
@@ -442,6 +442,7 @@ distclean: clean
 		html Makefile po/Makefile test/Makefile
 	$(RM) *.gc?? */*.gc?? test/*/*.gc??
 	$(RM) lcov.info lcov
+	$(RM) zzz.inc doxygen-build.txt tmp.txt
 
 # coverage testing
 coverage: all test
@@ -450,6 +451,11 @@ coverage: all test
 	lcov -t "result" -o lcov.info -c -d mutt -d config
 	genhtml -o lcov lcov.info
 
+# Web Code docs (doxygen)
+.PHONY: html $(PWD)/mutt
+html:
+	$(SRCDIR)/doxygen/generate.sh
+
 # Tests for the config code
 config-test: libmutt.a libemail.a libconfig.a libaddress.a
 
diff --git a/email/lib.h b/email/lib.h
index 6b086a8ea..7d050b8e3 100644
--- a/email/lib.h
+++ b/email/lib.h
@@ -41,6 +41,8 @@
  * | email/tags.c           | @subpage email_tags      |
  * | email/thread.c         | @subpage email_thread    |
  * | email/url.c            | @subpage email_url       |
+ *
+ * @image html libemail.svg
  */
 
 #ifndef MUTT_EMAIL_LIB_H
diff --git a/mutt/mutt.h b/mutt/mutt.h
index ffc53041f..9a0eb6b29 100644
--- a/mutt/mutt.h
+++ b/mutt/mutt.h
@@ -50,6 +50,8 @@
  *
  * @note The library is self-contained -- some files may depend on others in
  *       the library, but none depends on source from outside.
+ *
+ * @image html libmutt.svg
  */
 
 #ifndef MUTT_LIB_MUTT_H
