diff --git i/Makefile.autosetup w/Makefile.autosetup
index 359b32119..e6a065ef4 100644
--- i/Makefile.autosetup
+++ w/Makefile.autosetup
@@ -473,6 +473,7 @@ distclean: clean
 		test/Makefile .clang_complete
 	$(RM) *.gc?? */*.gc?? test/*/*.gc??
 	$(RM) lcov.info lcov
+	$(RM) zzz.inc doxygen-build.txt tmp.txt
 
 @if ENABLE_COVERAGE
 # coverage testing
@@ -488,6 +489,11 @@ coverage: all test
 	lcov --list lcov.info
 @endif
 
+# Web Code docs (doxygen)
+.PHONY: html $(PWD)/mutt
+html:
+	$(SRCDIR)/doxygen/generate.sh
+
 ##############################################################################
 # include generated dependency files
 -include $(DEPFILES)
diff --git i/email/lib.h w/email/lib.h
index 6b086a8ea..7d050b8e3 100644
--- i/email/lib.h
+++ w/email/lib.h
@@ -41,6 +41,8 @@
  * | email/tags.c           | @subpage email_tags      |
  * | email/thread.c         | @subpage email_thread    |
  * | email/url.c            | @subpage email_url       |
+ *
+ * @image html libemail.svg
  */
 
 #ifndef MUTT_EMAIL_LIB_H
diff --git i/mutt/mutt.h w/mutt/mutt.h
index abb8def80..1ca18e786 100644
--- i/mutt/mutt.h
+++ w/mutt/mutt.h
@@ -53,6 +53,8 @@
  *
  * @note The library is self-contained -- some files may depend on others in
  *       the library, but none depends on source from outside.
+ *
+ * @image html libmutt.svg
  */
 
 #ifndef MUTT_LIB_MUTT_H
